File: packages/php/src/Seguranca/AtaqueInterface.php
<?php

namespace Application\Seguranca;

interface AtaqueInterface
{
    public function nome(): string;

    /**
     * Roda o ataque contra o alvo e retorna o resultado.
     * Deve lançar SkipAtaqueException se o alvo não suportar os recursos
     * necessários (ex: não expõe gerarKeystreamBruto).
     */
    public function executar(AlvoCriptografico $alvo): ResultadoAtaque;
}

File: packages/java/src/main/java/fbc/Seguranca/AtaqueInterface.java
package fbc.Seguranca;

public interface AtaqueInterface {
    String nome();

    /**
     * Roda o ataque contra o alvo e retorna o resultado.
     * Deve lançar {@link SkipAtaqueException} se o alvo não suportar os recursos
     * necessários.
     */
    ResultadoAtaque executar(AlvoCriptografico alvo);
}
